src/components/Navbar.vue

```diff
@@ -70,20 +92,28 @@ export default {
         .then((response) => {
           if (response.status === 200) {
             this.numberOfCartItems = response.data.length;
+            this.token = Cookies.get("lemonToken");
           }
         })
         .catch((error) => {
           if (error.response.status === 401) {
             this.numberOfCartItems = 0;
+            this.token = "";
           }
         });
     },
+    toggle(event) {
+      this.$refs.menu.toggle(event);
+    },
   },
   created() {
     this.token = Cookies.get("lemonToken");
     this.getCart();
-    this.emitter.on("changeCartBadgeCount", (sizeOfCartItems) => {
-      this.numberOfCartItems = sizeOfCartItems;
+    this.emitter.on("changeCartBadgeCount", (numberOfCartItems) => {
+      this.numberOfCartItems = numberOfCartItems;
+    });
+    this.emitter.on("changeUserNavbarIconBtn", (token) => {
+      this.token = token;
```

Comment:
我覺得不用傳 token 進來，直接在這邊拿就好
另外用 IsLoggedIn 當變數來表示「是否有登入」會更清楚

```suggestion
    this.emitter.on("changeUserNavbarIconBtn", () => {
      const token = Cookies.get("lemonToken");
      if (token === null || token === undefined) {
        this.logged_in = false;
      } else {
        this.logged_in = true;
      }

```

Reply:
過期 token 一律清除 :  https://github.com/c711cat/lemon_coffee/pull/49/commits/fa21e22e1af76357e463f578f70c97193f8cc5b0
我可以把 this.logged_in 用原來就有的變數 this.token 沿用嗎? 
像這樣 : https://github.com/c711cat/lemon_coffee/pull/49/commits/b8333114b8c343c853fc98e552f9252814a9881d

Reply:
> 我可以把 this.logged_in 用原來就有的變數 this.token 沿用嗎?
像這樣 : b833311
 
OK